x-pack/plugins/entity_manager/server/lib/queries/index.ts

```diff
@@ -18,74 +19,149 @@ export const entitySourceSchema = z.object({
 
 export type EntitySource = z.infer<typeof entitySourceSchema>;
 
-const sourceCommand = ({ source }: { source: EntitySource }) => {
-  let query = `FROM ${source.index_patterns}`;
+const hasEntityTimestamp = (sources: EntitySource[]) => {
+  return sources.some((source) => source.timestamp_field);
+};
 
-  const esMetadataFields = source.metadata_fields.filter((field) =>
-    ['_index', '_id'].includes(field)
-  );
-  if (esMetadataFields.length) {
-    query += ` METADATA ${esMetadataFields.join(',')}`;
-  }
+const sourceCommand = ({
+  sources,
+  metadataFields,
+}: {
+  sources: EntitySource[];
+  metadataFields: string[];
+}) => {
+  const indexPatterns = sources.flatMap((source) => source.index_patterns).join(', ');
+  const esMetadataFields = ['_index', ...metadataFields.filter((field) => ['_id'].includes(field))];
+
+  return `FROM ${indexPatterns} METADATA ${esMetadataFields.join(', ')}`;
+};
+
+const sourcesEvalCommand = ({ sources }: { sources: EntitySource[] }) => {
+  const evals = sources.map((source, index) => {
+    const conditions = [
+      source.index_patterns
+        .map((pattern) => `_index LIKE "*${last(pattern.split(':'))}*"`)
```

Comment:
I don't recall the details, but there is a new ES syntax for index patterns where you can use `:` for CCS and `::` for something else, so this might cause issues here.